Cargamos las librerias necesarias y la base de datos

In [3]:
from sklearn.neighbors import KNeighborsRegressor
from sklearn.datasets import load_boston
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression, Ridge

In [5]:
boston = load_boston()

In [6]:
boston.keys()

dict_keys(['data', 'target', 'feature_names', 'DESCR', 'filename'])

tenemos de la base de datos 506 casas con 13 caractristicas  atributos por cada casa

In [14]:

boston.data.shape


(506, 13)

In [15]:
X_ent, X_test, y_ent, y_test=train_test_split(boston.data,boston.target)

In [16]:
X_ent.shape
X_test.shape

(127, 13)

Vamos a usar 379 datos para entrenar( usandoo _ent) y 127( usando _test) datos para que una vez aplicado el modelo veer si estan bien el modelo propuesto

In [19]:
knn = KNeighborsRegressor(n_neighbors=3)

In [20]:
knn.fit(X_ent,y_ent)

KNeighborsRegressor(algorithm='auto', leaf_size=30, metric='minkowski',
          metric_params=None, n_jobs=None, n_neighbors=3, p=2,
          weights='uniform')

¿Que tambien aprendio el modelo?

In [21]:
knn.score(X_test, y_test)

0.5243136168810236

El valor 0.52 es muy malo, un buen valor seria cercano a 1, para ello se propone cambiar el numero de vecinos cercanos, se habia colocado inincialmente un valor de 3

Aplicamos otro modelo  el de regresion y con .score vemos si es cercano a uno cuan bueno es

In [22]:
rl = LinearRegression()

In [23]:
rl.fit(X_ent,y_ent)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [24]:
rl.score(X_test,y_test)

0.7084929682450276

Aplicamos el modelo Ridge

In [27]:
ridge = Ridge(alpha=.2)
ridge.fit(X_ent,y_ent)
ridge.score(X_test,y_test)

0.7106487191141251